### ISSUE TITLE
Support HTTPS over HTTPS proxies

### ISSUE THREAD

**Issue by daurnimator** [daurnimator/lua-http#59]
Currently an NYI: https://github.com/daurnimator/lua-http/blob/87fed887592d387c79d666b7a0525ecfb4b34687/http/request.lua#L389

Links:
  - https://daniel.haxx.se/blog/2016/11/26/https-proxy-with-curl/
  - https://bugzilla.mozilla.org/show_bug.cgi?id=378637
  - https://dev.chromium.org/developers/design-documents/secure-web-proxy

**Comment by aleclarson** [comment]
Does this mean I can't put an HTTPS-enabled proxy (like nginx) in front of `lua-http`? When will this be supported?

**Comment by RyanSquared** [comment]
I think this means that `lua-http` itself can't be used as a proxy.

**Comment by daurnimator** [comment]
> Does this mean I can't put an HTTPS-enabled proxy (like nginx) in front of lua-http? 

No. This means that when making a request, you can't use an HTTPS outgoing proxy server and *also* use HTTPS inside of the request (and hence hiding the request from the proxy server). This is a very rare setup that even `curl` only started to support 2 years ago.
